### ISSUE TITLE
api : async issues

### ISSUE THREAD

**Issue by Firlfire** [vitaly-t/pg-promise#621]
Hi,

Sorry for this long post (I think the problem is that I don't use async/await and Promise as I should)
I'm implementing a json rpc API which use your pg-promise to manage a Database. I have some difficulties. Maybe I'm the problem and I do not use async/await and Promise as it should.
For the DB patterne I was inspired by your [pg-promise-demo](https://github.com/vitaly-t/pg-promise-demo)
I created a test file that send 10 000 insert requests to the API.

### Expected behavior
I want the server to be able to receive and process a large number of requests without making the client wait too long. I want these steps :

1. first: logs on the test console and the server console: the test script sends the requests and the server receives them as and when
2. then: logs on the server console and the test console: the server continues to receive the request while the client sends and at the same time the server inserts and sends the results when it can

### Actual behavior
For now, it works with 3 steps :

1. first: logs on the test console: sends requests
2. then: logs on the server console: when the test script finishes sending the requests, the server receives them ALL (the server runs until repos / roles.js line 27, then handles the next request)
3. then: logs on the server console and test script: the server executes each insert (call your function) and sends the results each time, and in parallel the test script displays the return on the console

Thank in advance! :D

### Steps to reproduce (some files)
**My test :**
```js
const request = require('request');
 
function concurrentRequests(){
  for (var count = 1; count <= 10000; count++) {
    let roleStatement = {
      jsonrpc: '2.0',
      host: 'test',
      id: count,
      method: "roles.insert",
      params: {
          return: true,
          data: { name: count.toString() }
        }
    };
 
    ContactApi(roleStatement)
      .then((result) => { Logger(`${JSON.stringify(result)}`); })
      .catch((ex) => { Logger(`ERROR ${roleStatement.id} : ${ex.message}`)});
  }
}
 
function ContactApi(statement) {
  return new Promise(function(resolve, reject) {
    Logger(`Prepare request for ${statement.method}(${statement.id})...`)
        request.post({
          url:'http://localhost:9000/api/v1',
          json: statement
        }, function(err,httpResponse,body){
          if(err) {
            reject(err);
          }
          else {
            resolve(body);
          }
        });
  });
}
 
function Logger(message) {
  const Timer = new Date();
  console.log(`${Timer} : ${JSON.stringify(message)}`);
}
 
concurrentRequests();
```

**the server :**
```js
const http = require('http');
const Database = require('../db'); //  ./db/index.js
 
const path = '/api/v1', port = 9000;
const methods = Database.methods;
 
/* hidden for brevity */
 
function sendResponse(res, response) {
  if (response) {
    const responseStr = JSON.stringify(response);
    res.setHeader('Content-Type', 'application/json');
    res.setHeader('Content-Length', responseStr.length);
    res.write(responseStr);
  } else {
    // Respond 204 for notifications with no response
    res.setHeader('Content-Length', 0);
    res.statusCode = 204;
  }
  res.end();
}
 
const requestHandler = (req, res) => {
 
  // lecture requete
  const body = [];
  req.on('data', (chunk) => {
    body.push(chunk);
  }).on('end', () => {
    const bodyStr = Buffer.concat(body).toString();
 
    res.setHeader('Content-Type', 'application/json');
 
    // parse body en JSON
    let request;
    try {
      request = JSON.parse(bodyStr);
    } catch (err) {
      const response = {
        id: null,
        jsonrpc: '2.0',
        error: {
          code: 'consts.PARSE_ERROR',
          message: err.message,
        },
      };
      sendResponse(res, response);
      return;
    }
 
    requestProcessor(request).then((response) => {
      sendResponse(res, response);
    });
  });
}
 
async function requestProcessor(request) {
  let response = {
    id: request.id,
    host: request.host,
    jsonrpc: '2.0',
  };
 
  try {
    response.result = await Promise.resolve(methods[request.method](request.params));
    if (!response.id) {
      response = undefined; // don't return a response if id is null
    }
  }
  catch (err) {
    /* hidden for brevity */
  }
 
  return response;
}
 
const server = http.createServer(requestHandler);
server.listen(port, (err) => {
  if(err) {
    return console.log('something bad happened', err);
  }
  console.log((`server is listening on ${port}`);
})
```

**One repo :**
```js
'use strict';
 
const cs = {}; // Reusable ColumnSet objects.
 
var Collections = {};
 
class RolesRepository {
  constructor(db, pgp) {
    this.db = db;
    this.pgp = pgp;
 
    // set-up all ColumnSet objects, if needed:
    this.Collections = createColumnsets(pgp);
 
    this.methods = {
        'roles.insert': this.add.bind(this)
    }
  }
 
  async add(params) {
    var query = this.pgp.helpers.insert(params.data, this.Collections.insert);
    if(params.return) query += " RETURNING *";
 
    Log(`inserting role ${JSON.stringify(params)}`);
    return this.db.any(query)
            .then(data => {
              Log(`INSERTED role ${JSON.stringify(data)}`);
              return data;
            })
            .catch(ex => {
              throw new Error(ex);
            });
  }
}
 
/* hidden for brevity */
 
module.exports = RolesRepository;
```

### Environment

* Version of pg-promise: 8.5.1
* OS type (Linux/Windows/Mac): Windows
* Version of Node.js: 10.15.1

**Comment by vitaly-t** [comment]
If you were to open such question on StackOverflow, even there you would be asked to localize the issue before asking, but here it is completely out of scope, investigating issues in your app architecture.
